stats/kernel-selftests

```diff
@@ -1112,37 +1112,50 @@ class NetfilterStater < Stater
       @test_case = $1 if @test_script == 'nft_concat_range.sh'
     when /^#   (.+)\[( OK|FAIL|SKIP)/
       #   Add two elements, flush, re-add                               [ OK ]
-      @subtest_case = $1.strip if @test_script == 'nft_concat_range.sh'
-      stats.add "#{@test_prefix}.#{@test_case}.#{$1}", $2 if @test_script == 'nft_concat_range.sh'
+      test_case = $1
+      result = $2
+      parse_subtest_result(test_case, result, stats)
     when /^#     (baseline .+|set with .+):\s*(\S*)pps/
       #     baseline (drop from netdev hook):               1612678pps
       stats.add "#{@test_prefix}.#{@test_case}.#{@subtest_case}.#{$1}.packets_per_sec", $2.to_i if @test_script == 'nft_concat_range.sh'
-    when /^# (PASS|FAIL|SKIP): (ns\d)-\S+ (connection on port.*)/
+    when /^# (PASS|FAIL|SKIP|OK): (.*)/
+      test_case = $2
+      result = $1
+      parse_test_result(test_case, result, stats)
+    else
+      super(line, stats)
+    end
+  end
+
+  def parse_subtest_result(test_case, result, stats)
+    return unless @test_script == 'nft_concat_range.sh'
+
+    @subtest_case = test_case.strip
+    stats.add "#{@test_prefix}.#{@test_case}.#{test_case}", result
+  end
+
+  def parse_test_result(test_case, result, stats)
+    if test_case =~ /(ns\d)-\S+ (connection on port.*)/
```

Comment:
prefer to match different line outside of this function, so in the function, better to only wrap the logic of "when /^# (PASS|FAIL|SKIP|OK): (.*)/" part.

Also two when clauses are slightly different, one can match OK, one can't. Thus combine it to match (PASS|FAIL|SKIP|OK): (.*) may bring some confusion.